### PR TITLE
☠️ Fatal Scenario: Extra Metadata

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're swapping `abort()` with an actual NSException (with extra payload), to be thrown whenever there are problems initializing the NSPersistentStoreCoordinator (or the ApplicationSupport folder cannot be created).

Our goal is to add extra metadata, that might help us track down a a crasher affecting the D1 app.

[Sample of how this looks like in Sentry](https://sentry.io/organizations/a8c/issues/2857939405/events/0996f6bb520b4f09a42829418d2424d2/?project=5959376)

cc @jkmassel WDYT sir?

### CODE REVIEW

File: Sources/Model/ObjC/Common/Core Data/TracksContextManager.m [hunk 1]
```diff
@@ -82,12 +93,25 @@ - (NSURL *)applicationSupportURLForAppWithBundleIdentifier:(NSString *)bundleIde
...
+            abort();
+        }
     }
 
     return folder;
 }
 
+- (BOOL)shouldThrowUponFatalCondition {
+    // We consider it safe to Throw (whenever a Fatal Condition arises) whenever there is no global
+    // NSException handler set
```
Comment (MichelleTessier) [inline Sources/Model/ObjC/Common/Core Data/TracksContextManager.m:111]: Whoa, I am no objc exception expert! Can you give me a rundown of how this is working @jleandroperez ? What is `NSGetUncaughtExceptionHandler()`?

File: Sources/Model/ObjC/Common/Core Data/TracksContextManager.m [hunk 2]
```diff
@@ -82,12 +93,25 @@ - (NSURL *)applicationSupportURLForAppWithBundleIdentifier:(NSString *)bundleIde
     // available and one should always be able to create a folder in it
     if (error != nil) {
         TracksLogError(@"Failed to create folder for %@ in Application Support: %@, %@", bundleIdentifier, error, [error userInfo]);
-        abort();
+
+        if (self.shouldThrowUponFatalCondition) {
+            @throw [NSException exceptionWithName:TracksApplicationSupportException
```
Comment (MichelleTessier) [inline Sources/Model/ObjC/Common/Core Data/TracksContextManager.m:98]: Do these get caught somewhere? I need a rundown on objc exceptions @bjhomer.

**Review by bjhomer — APPROVED** [review]
This looks good to me. I normally don't love using ObjC exceptions, but in this case we're still intending to crash; we're just doing so with a bit more context.

**Comment by jleandroperez** [comment]
Thank you BJ!!

**Pull request merged**